Sources/TuistCore/MetadataProviders/XCFrameworkSignatureProvider.swift

```diff
@@ -0,0 +1,99 @@
+import Command
+import CryptoKit
+import FileSystem
+import Foundation
+import Path
+import TSCBasic
+import TuistSupport
+import XcodeGraph
+
+enum XCFrameworkSignatureProviderError: LocalizedError, Equatable {
+    case codesignOutputMissing
+    case certificateFileReadFailed
+    case appleSignedXCFrameworkMissingDetails(teamIdentifier: String?, teamName: String?)
+
+    var errorDescription: String {
+        switch self {
+        case .codesignOutputMissing:
+            return "codesign finished, but no output file was found."
+        case .certificateFileReadFailed:
+            return "Failed to read certificate file."
+        case let .appleSignedXCFrameworkMissingDetails(teamIdentifier, teamName):
+            return "Apple signed XCFramework missing team identifier or name. teamIdentifier: \(teamIdentifier ?? "nil"), teamName: \(teamName ?? "nil")"
```

Comment:
I'd improve the wording here and include the path to the XCFramework to ease debugging.

```suggestion
            switch (teamIdentifier, teamName) {
              (nil, nil): return "The framework at \(path) signed with an Apple certificate lacks the team identifier and name."
              (nil, _): return "The framework at \(path) signed with an Apple certificate lacks the team identifier."
              (_, nil): return "The framework at \(path) signed with an Apple certificate lacks the team name."
            }
```